### PR TITLE
Fix array macros for Ember.Array objects

### LINKED ISSUE #423: Array macros fail with Array-like objects
Ember supports objects that implement the array API but don't descend from the native `Array` class. When you use Ember ways to work with arrays, such fake arrays are indistinguishable from real arrays.

The most common example is Ember Data's synchronous relationships:

```js
export default DS.Model.extend({
  children: DS.hasMany('child', {async: false})
})
```
The awesome utils fail to recognize a fake array as an array and yield absolutely incorrect results.

This issue comes with a failing test case and a PR that resolves it, in separate commits.

### PR DESCRIPTION
**Pull request by lolmaus**
Resolves #423 

A failing test case is in a separate [commit](https://github.com/kellyselden/ember-awesome-macros/commit/1e55173a36847f783b14c9d942e49ad7c0e9da03) for your convenience.

### CODE REVIEW

**Comment by Turbo87** [comment]
@lolmaus thanks for the PR. I've rebased it and hopefully @kellyselden will be able to review tomorrow :)

File: tests/unit/many-array-test.js [hunk 1]
```diff
@@ -0,0 +1,233 @@
+import { moduleForModel, test } from 'ember-qunit';
+import compute from 'ember-macro-test-helpers/compute';
+import { getBy } from 'ember-awesome-macros';
+import array from 'ember-awesome-macros/array';
+import raw from 'ember-macro-helpers/raw';
+import { run } from '@ember/runloop';
+
+let parent, child1, child2;
+
+moduleForModel('parent', 'Unit | ManyArray', {
```
Comment (kellyselden) [inline tests/unit/many-array-test.js]: Looks like the rest of the app now uses the new test syntax. Can you run the conversion here too please?

File: tests/unit/many-array-test.js [hunk 2]
```diff
@@ -0,0 +1,233 @@
+import { moduleForModel, test } from 'ember-qunit';
```
Comment (kellyselden) [inline tests/unit/many-array-test.js]: This seems kind of like an acceptance test. Maybe it should move there, I'm not sure.

Comment (Turbo87) [reply tests/unit/many-array-test.js]: agreed, moved

File: tests/unit/many-array-test.js (hunk 1 above)
Comment (Turbo87) [reply tests/unit/many-array-test.js]: yep, done :)

**Pull request merged**